src/Features/Core/Portable/MetadataAsSource/DecompilationMetadataAsSourceFileProvider.cs

```diff
@@ -172,13 +195,16 @@ public DecompilationMetadataAsSourceFileProvider(IImplementationAssemblyLookupSe
                     }
                 }
 
-                using (var textWriter = new StreamWriter(fileInfo.TemporaryFilePath, append: false, encoding: MetadataAsSourceGeneratedFileInfo.Encoding))
+                if (!skipWritingFile)
                 {
-                    text.Write(textWriter, cancellationToken);
-                }
+                    using (var textWriter = new StreamWriter(fileInfo.TemporaryFilePath, append: false, encoding: MetadataAsSourceGeneratedFileInfo.Encoding))
+                    {
+                        text.Write(textWriter, cancellationToken);
+                    }
 
-                // Mark read-only
-                new FileInfo(fileInfo.TemporaryFilePath).IsReadOnly = true;
+                    // Mark read-only
+                    new FileInfo(fileInfo.TemporaryFilePath).IsReadOnly = true;
```

Comment:
Why are we creating a new FileInfo w/o using it?

Reply:
We are using the `IsReadOnly` setter. 

➡️ This code did not change.

Reply:
> This code did not change

Yea, I see that, just wondering if we should use this new FileInfo instead afterward.